src/h5web/vis-packs/core/heatmap/ShadersMesh.tsx

```diff
@@ -0,0 +1,79 @@
+import { useMemo, ReactElement, memo } from 'react';
+import { useThree } from 'react-three-fiber';
+import { DataTexture, FloatType, RedFormat } from 'three';
+import type { Domain } from '../models';
+
+interface Props {
+  rows: number;
+  cols: number;
+  values: number[];
+  domain: Domain;
+}
+
+function ShadersMesh(props: Props): ReactElement {
+  const { rows, cols, values, domain } = props;
+
+  const shader = useMemo(
+    () => ({
+      uniforms: {
+        data: {
+          value: new DataTexture(
+            Float32Array.from(values),
+            cols,
+            rows,
+            RedFormat,
```

Comment:
`RedFormat` avoids having to add three zeros for each value to our `values` array before passing it to the `DataTexture`.